images/bridge2/workingset/commands.go

```diff
@@ -0,0 +1,105 @@
+package workingset
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"net"
+
+	"github.com/ajbouh/substrate/images/bridge2/tracks"
+	"github.com/ajbouh/substrate/pkg/commands"
+)
+
+type CommandProvider struct{}
+
+func (c *CommandProvider) Commands(sess *tracks.Session) commands.Source {
+	src := &commands.DynamicSource{
+		Sources: []commands.Source{
+			commands.NewStaticSource([]commands.Entry{
+				{Name: "add_url",
+					Def: commands.Def{
+						Description: `add_url(url: str) -> bool`,
+						Parameters: commands.FieldDefs{
+							"url": {
+								Name:        "url",
+								Type:        "string",
+								Description: "URL to add to working set",
+							},
+						},
+						Returns: nil,
+					},
+					Run: func(ctx context.Context, args commands.Fields) (commands.Fields, error) {
+						url := args.String("url")
+						AddURL(sess.SpanNow(), url)
+						return commands.Fields{
+							"success": true,
+						}, nil
+					},
+				},
+
+				{Name: "list",
+					Def: commands.Def{
+						Description: `list() -> []string`,
+						Parameters:  commands.FieldDefs{},
+						Returns:     nil,
+					},
+					Run: func(ctx context.Context, args commands.Fields) (commands.Fields, error) {
+						urls := ActiveURLs(sess)
+						return commands.Fields{
+							"urls": urls,
+						}, nil
+					},
+				},
+			}),
+		},
+	}
+	for _, url := range ActiveURLs(sess) {
+		src.Sources = append(src.Sources, &commands.PrefixedSource{
```

Comment:
My sense here is that every entry in workingset should be under a human readable key. Consider a situation where a user intentionally adds the same URL twice. We'd prefer that the namespace for those two pages stays separate.